addons/udes_suggest_location/tests/test_stock_move_line.py

```diff
@@ -234,3 +236,276 @@ def test_set_invalid_dest_location(self):
+        cls.create_quant(cls.apple.id, cls.test_stock_location_01.id, 10)
+        cls.create_quant(cls.banana.id, cls.test_stock_location_02.id, 10)
+        cls.create_quant(cls.cherry.id, cls.test_stock_location_03.id, 10)
+        cls.create_quant(cls.damson.id, cls.test_stock_location_04.id, 10)
+        cls.create_quant(cls.apple.id, cls.test_stock_location_01.id, 10)
+
+        # Create picking
+        origin = 'SO0001'
+        cls._pick_info1 = [{"product": cls.apple, "qty": 10}]
+        cls._pick_info2 = [{"product": cls.banana, "qty": 10}, {"product": cls.cherry, "qty": 10}]
+        cls._pick_info3 = [{"product": cls.damson, "qty": 10}]
+        cls.picking1 = cls.create_picking(
+            cls.picking_type_pick,
+            products_info=cls._pick_info1,
+            origin=origin,
+            confirm=True,
+            assign=True,
+        )
+        cls.picking2 = cls.create_picking(
+            cls.picking_type_pick,
+            products_info=cls._pick_info2,
+            origin=origin,
+            confirm=True,
+            assign=True,
+        )
+        cls.picking3 = cls.create_picking(
+            cls.picking_type_pick,
+            products_info=cls._pick_info3,
+            origin='SO0002',
+            confirm=True,
+            assign=True,
+        )
+        cls.mls = cls.picking1.move_line_ids | cls.picking2.move_line_ids | cls.picking3.move_line_ids
+
+        # Complete pickings 1 & 3. Picking 2 should have Check 01 suggested, as
+        # it shares an origin with Picking 1.
+        for picking, location in zip((cls.picking1, cls.picking3), (cls.test_check_location_01, cls.test_check_location_02)):
+            picking.move_line_ids.location_dest_id = location
+            picking.move_line_ids.qty_done = picking.move_line_ids.product_uom_qty
+            picking._action_done()
+
+        # Destination locations for each product
+##        cls.pick_apple_locs = cls.test_check_location_01 | cls.test_check_location_03
+        cls.pick_banana_locs = cls.test_check_location_01
+        pick_empty_locs = cls.test_check_location_03 | cls.test_check_location_04
+        cls.pick_all_locs = cls.pick_banana_locs | pick_empty_locs
+
+        # Particular movelines for pick
+        cls.apple_mls = cls.mls.filtered(lambda ml: ml.product_id == cls.apple)
+        cls.banana_mls = cls.mls.filtered(lambda ml: ml.product_id == cls.banana)
+        cls.cherry_mls = cls.mls.filtered(lambda ml: ml.product_id == cls.cherry)
+
+##        # Mock functions, for suggest locations and validate
+##        # Suggest locations return empty locations
+##        cls.mock_suggested_locations = patch.object(
+##            cls.MoveLine.__class__,
+##            "suggest_locations",
+##            return_value=cls.test_check_location_04,
+##        )
+##        cls.mock_validate_location_dest = patch.object(
+##            cls.MoveLine.__class__, "validate_location_dest", return_value=None
+##        )
```

Comment:
Still need these?

Reply:
I don't know yet - that will be part of the tests clean up.